src/charm.py

```diff
@@ -1,69 +1,159 @@
         self._add_mongodb_org_repository()
         self._install_apt_packages(["mongodb-org"])
-        self.unit.status = WaitingStatus("Waiting to start MongoDB")
 
-    def _add_mongodb_org_repository(self):
+    def _on_config_changed(self, _: ops.charm.ConfigChangedEvent) -> None:
+        """Event handler for configuration changed events."""
+        # TODO
+        # - update existing mongo configurations based on user preferences
+        # - add additional configurations as according to spec doc
+        with open("/etc/mongod.conf") as mongo_config_file:
+            mongo_config = yaml.safe_load(mongo_config_file)
+
+        machine_ip = str(self.model.get_binding(PEER).network.bind_address)
+        bind_ips = "localhost," + machine_ip
+        mongo_config["net"]["bindIp"] = bind_ips
+        if "replication" not in mongo_config:
+            mongo_config["replication"] = {}
+
+        mongo_config["replication"]["replSetName"] = "rs0"
+        with open("/etc/mongod.conf", "w") as mongo_config_file:
+            yaml.dump(mongo_config, mongo_config_file)
+
+        self.unit.status = ops.model.ActiveStatus()
+
+    def _on_start(self, event: ops.charm.StartEvent) -> None:
+        """Enables MongoDB service and initializes replica set.
+
+        Args:
+            event (ops.charm.StartEvent): The triggering start event.
+        """
+        if not self.unit.is_leader():
+            return
+
+        # start mongo service
+        self.unit.status = MaintenanceStatus("enabling MongoDB")
+        if not service_running("mongod.service"):
+            logger.debug("enabling MongoDB")
+            mongod_enabled = service_resume("mongod.service")
+            if not mongod_enabled:
+                logger.error("failed to enable mongod.service")
+                self.unit.status = BlockedStatus("failed to enable mongod.service")
+                return
+
+        try:
+            self._open_port_tcp(self._port)
+        except Exception:
+            self.unit.status = BlockedStatus("failed to open TCP port for MongoDB")
+            return
+
+        if not self.mongo.is_ready():
+            self.unit.status = WaitingStatus("Waiting for MongoDB Service")
+            event.defer()
+            return
+
+        # initialize replica set
+        self.unit.status = MaintenanceStatus("initializing MongoDB replicaset")
+        logger.debug("initalizing replica set")
+        try:
+            logger.debug("initialzing replica set for these IPs %s", self._unit_ips)
+            self.mongo.initialize_replica_set(self._unit_ips)
+            self._peers.data[self.app]["replica_set_hosts"] = json.dumps(self._unit_ips)
+        except Exception as e:
+            logger.error("Error initializing replica sets in _on_start: error={}".format(e))
+            self.unit.status = BlockedStatus("failed to initialize replicasets")
+            return
+
+        self.unit.status = ActiveStatus("MongoDB started")
```

Comment:
```suggestion
        self.unit.status = ActiveStatus()
```

In general, avoid putting a message in with `ActiveStatus`